omod/src/main/java/org/openmrs/module/referenceapplication/page/controller/LoginPageController.java

```diff
@@ -206,24 +227,43 @@ public String post(@RequestParam(value = "username", required = false) String us
 			}
 		}
 
-		//TODO uncomment this to replace the if clause after it
-		if (sessionLocation != null && sessionLocation.hasTag(EmrApiConstants.LOCATION_TAG_SUPPORTS_LOGIN)) {
-			// Set a cookie, so next time someone logs in on this machine, we can default to that same location
-			Cookie cookie = new Cookie(COOKIE_NAME_LAST_SESSION_LOCATION, sessionLocationId.toString());
-			cookie.setHttpOnly(true);
-			pageRequest.getResponse().addCookie(cookie);
+		try {
+			Context.authenticate(username, password);
+			String locationUserPropertyName = administrationService.getGlobalProperty(ReferenceApplicationConstants.LOCATION_USER_PROPERTY_NAME);
+			if (StringUtils.isNotBlank(locationUserPropertyName)) {
+				if (Context.isAuthenticated() && Context.getUserContext().getAuthenticatedUser() != null) {
+					String locationUuid = Context.getUserContext().getAuthenticatedUser().getUserProperty(locationUserPropertyName);
+					if (StringUtils.isNotBlank(locationUuid)) {
+						sessionLocation = locationService.getLocationByUuid(locationUuid);
+					}
+					if (sessionLocation != null) {
+						sessionLocationId = sessionLocation.getLocationId();
+                    }
+					else {
```

Comment:
The formatting or braces around here do not look correct.

Reply:
Fixed